v_frame/src/plane.rs

```diff
@@ -473,6 +484,79 @@ impl<T: Pixel> Plane<T> {
     new
   }
 
+  /// Returns plane with downscaled resolution
+  /// Downscaling the plane by integer value
+  /// Not padded
+  #[hawktracer(downscale)]
+  pub fn downscale(&self, scale: usize) -> Plane<T> {
+    let box_pixels = scale * scale;
+    let half_box_pixels = box_pixels as u32 / 2; // Used for rounding int division
+
+    let src = self;
+    let data_origin = src.data_origin();
+
+    // unsafe: all pixels initialized in this function
+    let mut new_plane = unsafe {
+      Plane::new_uninitialized(
+        src.cfg.width / scale,
+        src.cfg.height / scale,
+        0,
+        0,
+        0,
+        0,
+      )
+    };
+
+    let stride = new_plane.cfg.stride;
+    let width = new_plane.cfg.width;
+    let height = new_plane.cfg.height;
+
+    // Par iter over dst chunks
+    let np_raw_slice = new_plane.data.deref_mut();
+    let threads = current_num_threads();
+    let chunk_rows = cmp::max((height + threads / 2) / threads, 1);
+
+    let chunk_size = chunk_rows * stride;
+
+    let height_limit = height * stride;
+    np_raw_slice[0..height_limit]
+      .par_chunks_mut(chunk_size)
+      .enumerate()
+      .for_each(|(chunk_idx, chunk)| {
+        // Iter dst rows
+        let dst_rows = chunk.chunks_mut(stride);
+        for (row_offset, dst_row) in dst_rows.enumerate() {
+          let row_idx = chunk_idx * chunk_rows + row_offset;
+
+          // Iter dst cols
+          for (col_idx, dst) in dst_row[0..width].iter_mut().enumerate() {
+            let mut sum = half_box_pixels;
+
+            // Sum box of size scale * scale
+            // TODO: use SIMD here, maybe try `faster` crate
```

Comment:
Remove the reference to `faster` please.

Reply:
Removed